scala-package/core/src/main/scala/org/apache/mxnet/NDArray.scala

```diff
@@ -563,6 +568,61 @@ class NDArray private[mxnet](private[mxnet] val handle: NDArrayHandle,
     NDArrayCollector.collect(this)
   }
 
+  /**
+    * Java Flavor creating new NDArray
+    * @param arr
+    * @param shape
+    * @param ctx
+    * @return
+    */
+  def this(arr : Array[Float], shape : Shape, ctx : Context) = {
+    this(NDArray.newAllocHandle(shape, ctx, delayAlloc = false, Base.MX_REAL_TYPE))
+    this.set(arr)
+  }
+
+  override def toString: String = {
+    buildStringHelper(this, this.shape.length) + "\n"
```

Comment:
print NDArray's metadata instead
(x.shape, x.size, x.dtype, x.context, x.handle(%x)->hex value of native mem ref)